internal/db/migrations/007_create_gcp_reservation_details_table.sql

```diff
@@ -0,0 +1,14 @@
+
+CREATE TABLE gcp_reservation_details
+(
+  reservation_id BIGINT NOT NULL,
+  provider INTEGER NOT NULL DEFAULT provider_type_gce(),
+  pubkey_id BIGINT NOT NULL REFERENCES pubkeys(id),
+  source_id TEXT NOT NULL,
+  image_id TEXT NOT NULL,
+  gcp_reservation_id TEXT,
+  detail JSONB,
+
+  FOREIGN KEY (reservation_id, provider) REFERENCES reservations(id, provider) ON DELETE CASCADE,
+  CHECK (provider = provider_type_gce())
+);
```

Comment:
Since it is not as easy as before to do SQL DDL, let's think twice about the `gcp_reservation_id`, that is probably not the name we end up using. Can you figure out how the "job system" of GCP work and what identifier are we getting? I slightly looked into this and could not figure it out quickly, so I gave up :-)